Sources/PackageModel/ManifestSourceGeneration.swift

```diff
@@ -0,0 +1,326 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import TSCBasic
+import TSCUtility
+import Foundation
+
+/// Extensions on Manifest for generating source code expressing its contents
+/// in canonical declarative form.  Note that this bakes in the results of any
+/// algorithmically generated manifest content, so it is not suitable for the
+/// mechanical editing of package manifests.  Rather, it is intended for such
+/// tasks as manifest creation as part of package instantiation, etc.
+extension Manifest {
+    
+    /// Generates and returns a string containing the contents of the manifest
+    /// in canonical declarative form.
+    public var generatedManifestFileContents: String {
+        return """
+            // swift-tools-version: \(toolsVersion.description)
```

Comment:
Using `description` isn't good enough here; I forgot to add a FIXME in the code.

Reply:
I think just toolsVersion should be enough. I'm not completely sure, but I think since ToolsVersion conforms to CustomStringConvertible, its instances can be interpolated directly.

Additionally, I'm wondering if this could use toolsVersionWriter? toolsVersionWriter currently prepends the "swift-tools-version" line to the manifest (or replace the line when the tools version is invalid), so it seems like it might be a good fit here?

Reply:
I wasn't aware of that, but it sounds like a good fit.  I'll look into it.  Thanks!

Reply:
`toolsVersionWriter` looks promising but seems to rely on a file system.  If this were split out into a ToolsVersionFormatter, that would be perfect, and `toolsVersionWriter` were written in terms of that, that would be great.

Reply:
Also, the `Format.asJSON(version.major)` and `Format.asJSON(version.minor)` look a little odd.  These version numbers are not quoted or processed in other ways (and there is no connection to JSON), so that code looks a little suspicious.

Reply:
> If this were split out into a ToolsVersionFormatter, that would be perfect, and `toolsVersionWriter` were written in terms of that, that would be great.

This makes sense, and I kind of want to take it on myself once #2937 (which modifies the file) settles. 

> Also, the `Format.asJSON(version.major)` and `Format.asJSON(version.minor)` look a little odd. These version numbers are not quoted or processed in other ways (and there is no connection to JSON), so that code looks a little suspicious.

Yes `asJSON` is quite odd. [It actually doesn't do anything.](https://github.com/apple/swift-tools-support-core/blob/8123987b1681af3db7b695d05627d58c979809f3/Sources/TSCBasic/WritableByteStream.swift#L445) So it's probably safe to just replace it with string interpolation.